spec/views/schools/ects/show.html.erb_spec.rb

```diff
@@ -0,0 +1,131 @@
+    end
+
+    it 'full name' do
+      expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'Name')
+      expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'Barry White')
+    end
+
+    it 'email address' do
+      expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'Email address')
+      expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'love@whale.com')
+    end
+
+    describe 'mentor' do
+      context 'when assigned' do
+        before do
+          mentor = FactoryBot.create(:teacher, trs_first_name: 'Moby', trs_last_name: 'Dick')
+          mentor_at_school_period = FactoryBot.create(:mentor_at_school_period, :active, school: school_2, teacher: mentor)
+
+          FactoryBot.create(:mentorship_period, :active,
+                            started_on: current_ect_period.started_on,
+                            mentee: current_ect_period,
+                            mentor: mentor_at_school_period)
+
+          render
+        end
+
+        it "has mentor's full name" do
+          expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'Mentor')
+          expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'Moby Dick')
+        end
+      end
+
+      context 'when unassigned' do
+        it 'has instruction to assign' do
+          expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'Mentor')
+          expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'You must assign a mentor or register a new one for this ECT.')
+        end
+      end
+    end
+
+    it 'school start date' do
+      expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'School start date')
+      expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'January 2025')
+    end
+
+    it 'working pattern' do
+      expect(rendered).to have_css('dt.govuk-summary-list__key', text: 'Working pattern')
+      expect(rendered).to have_css('dd.govuk-summary-list__value', text: 'Full time')
+    end
+  end
+
+  describe 'current training details summary card' do
+    before do
+      academic_year = FactoryBot.create(:academic_year)
+      lead_provider = FactoryBot.create(:lead_provider, name: 'Ambition institute')
+      delivery_partner = FactoryBot.create(:delivery_partner)
+      provider_partnership = FactoryBot.create(:provider_partnership, lead_provider:, delivery_partner:, academic_year:)
+      appropriate_body = FactoryBot.create(:appropriate_body, name: 'Alpha Teaching School Hub')
+
+      FactoryBot.create(:training_period, :active, :for_ect, provider_partnership:,
+                                                             ect_at_school_period: current_ect_period,
+                                                             started_on: current_ect_period.started_on)
+
+      FactoryBot.create(:induction_period, :active, teacher:, appropriate_body:)
```

Comment:
Do we need to create these records or will building work?

Reply:
I will find out

Reply:
No